src/dishka/entities/key.py

```diff
@@ -5,8 +5,15 @@
 from .component import DEFAULT_COMPONENT, Component
 
 
-class FromComponent(NamedTuple):
-    component: Component = DEFAULT_COMPONENT
+class _FromComponent(NamedTuple):
```

Comment:
how about using a frozen dataclass or a concrete class with a ctor, this isnt exactly a actual tuple in use, its more like a stash type to add a annotation